bin/desi_qso_qn_afterburner

```diff
@@ -0,0 +1,376 @@
+                        filename_priors, filename_output_rerun_RR, filename_zbest_rerun_RR
+
+    Returns:
+        redshift (numpy array): Array containing best redshift estimation by the new run of RR
+        err_redshift (numpy array): Array containing the associated error for the redshift
+        coeffs (numpy array): array containing the coefficient for the best fit given by RR
+                       even we work only with QSO template, it has a "shape" of redshift.size x 10
+                        WARNING: they have to be converted into a list (with .tolist()) to be added in the pandas dataframe
+    """
+
+
+    template_filename = param_RR['template_filename']
+    filename_priors =  param_RR['filename_priors']
+    filename_output_rerun_RR =  param_RR['filename_output_rerun_RR']
+    filename_zbest_rerun_RR = param_RR['filename_zbest_rerun_RR']
+
+    def write_prior_for_RR(targetid, z_prior, filename_priors):
+        # Write the prior file for RR associated to the targetid list
+        # Args:
+        #    targetid (int array): array of the targetid
+        #                          on which RR will be rerun with prior and qso template.
+        #    z_prior (float array): array of the same size than targetid with the
+        #                           redshift estimated by QN for the associated targetid
+        #    filename_priors (str): name under which the file will be saved
+
+        function = np.array(['tophat'] * z_prior.size) # need to be the same for every one
+        sigma = 0.1*np.ones(z_prior.size)
+        # save
+        out = fitsio.FITS(filename_priors, 'rw', clobber=True)
+        data, names, extname = [targetid, function, z_prior, sigma], ['TARGETID', 'FUNCTION', 'Z', 'SIGMA'], 'PRIORS'
+        out.write(data, names=names, extname=extname)
+        out.close()
+        log.debug(f'Write prior file for RR with {z_prior.size} objetcs: {filename_priors}')
+        return
+
+    def extract_redshift_info_from_RR(filename_zbest):
+         # extract information of the zbest file from the new RR run
+         # Args:
+         #    filename_zbest (str): Name of the zbest file from the new run of RR
+         # Returns:
+         #    redshift (numpy array): Array containing best redshift estimation by the new run of RR
+         #    err_redshift (numpy array): Array containing the associated error for the redshift
+         #    coeffs (numpy array): array containing the coefficient for the best fit given by RR
+         #                   even we work only with QSO template, it has a "shape" of redshift.size x 10
+         #                    WARNING: they have to be converted into a list (with .tolist()) to be added in the pandas dataframe
+
+        with fitsio.FITS(filename_zbest) as zbest:
+            redshift = zbest[1]['Z'][:]
+            err_redshift = zbest[1]['ZERR'][:]
+            coeffs = np.zeros((redshift.size, 10))
+            coeffs[:, :4] = zbest[1]['COEFF'][:]
```

Comment:
Please read DESI FITS file HDUs by name not number

Reply:
changed